internal/util/util.go

```diff
@@ -158,9 +158,11 @@ func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1api.VolumeSnap
 		return vsc, nil
 	}
 
-	// We'll wait 10m for the VSC to be reconciled polling every 5s
-	// TODO: make this timeout configurable.
+	// We'll wait 10m for the VSC to be reconciled polling every 5s unless csiSnapshotTimeout is set
 	timeout := 10 * time.Minute
+	if csiSnapshotTimeout > 0 {
```

Comment:
should we do null checks before this?

Reply:
https://github.com/vmware-tanzu/velero/blob/25fb08b3c27955cfb04ab72037faa996a117c8a7/pkg/controller/backup_controller.go#LL330C2-L333C3

You might want to add default case handling similar to the core code polling.